app/src/main/java/com/nononsenseapps/feeder/ui/compose/settings/OpenAISection.kt

```diff
@@ -0,0 +1,332 @@
+        ) {
+            Text(
+                text = stringResource(R.string.api_key),
+                style = MaterialTheme.typography.titleMedium,
+            )
+            Spacer(Modifier.weight(1.0f))
+            IconButton(onClick = {
+                onEvent(OpenAISettingsEvent.UpdateSettings(current))
+                onEvent(OpenAISettingsEvent.SwitchEditMode(enabled = false))
+            }) {
+                Icon(Icons.Outlined.Save, contentDescription = stringResource(R.string.save))
+            }
+
+            IconButton(onClick = { onEvent(OpenAISettingsEvent.SwitchEditMode(enabled = false)) }) {
+                Icon(Icons.Outlined.Close, contentDescription = stringResource(android.R.string.cancel))
+            }
+        }
+        TextField(
+            modifier = Modifier.fillMaxWidth(),
+            value = current.key,
+            onValueChange = {
+                current = current.copy(key = it)
+                onEvent(OpenAISettingsEvent.LoadModels)
+            },
+            visualTransformation = VisualTransformationApiKey(),
+        )
+        Text(
+            modifier = Modifier.padding(top = 8.dp),
+            text = stringResource(R.string.model_id),
+            style = MaterialTheme.typography.titleMedium,
+        )
+        TextField(
+            modifier = Modifier.fillMaxWidth(),
+            value = current.modelId,
+            onValueChange = {
+                current = current.copy(modelId = it)
+                onEvent(OpenAISettingsEvent.LoadModels)
+            },
+            trailingIcon = {
+                IconButton(
+                    onClick = { modelsMenuExpanded = true },
+                    enabled = models is OpenAIModelsState.Success,
+                ) {
+                    if (models is OpenAIModelsState.Loading) {
+                        CircularProgressIndicator()
+                    } else {
+                        Icon(Icons.Filled.ExpandMore, contentDescription = stringResource(R.string.list_of_available_models))
+                    }
+                }
+            },
+        )
+        OpenAIModelsSection(
+            menuExpanded = modelsMenuExpanded,
+            state = models,
+            onValueChange = {
+                current = current.copy(modelId = it)
+            },
+            onDismissRequest = { modelsMenuExpanded = false },
+        )
+
+        Text(
+            modifier = Modifier.padding(top = 8.dp),
+            text = stringResource(R.string.url),
+            style = MaterialTheme.typography.titleMedium,
+        )
+        TextField(
+            modifier = Modifier.fillMaxWidth(),
+            value = current.baseUrl,
+            placeholder = {
+                Text(OpenAIHost.OpenAI.baseUrl)
+            },
+            onValueChange = {
+                current = current.copy(baseUrl = it)
+                onEvent(OpenAISettingsEvent.LoadModels)
+            },
+        )
```

Comment:
```suggestion
        TextField(
            modifier = Modifier.fillMaxWidth(),
            value = current.baseUrl,
            placeholder = {
                Text(OpenAIHost.OpenAI.baseUrl)
            },
            label = {
                Text(text = stringResource(R.string.url))
            },
            onValueChange = {
                current = current.copy(baseUrl = it)
                onEvent(OpenAISettingsEvent.LoadModels)
            },
        )
```

You should not make separate Text fields for labels. Use the built-in label in the TextField. I'll only comment on this but I mean for all of them.

Also let the surrounding column deal with padding with `verticalArrangement` instead of adding individual padding to each each textfield.